bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/characteristics/node/PCMNodeCharacteristicsCalculatorImpl.java

```diff
@@ -0,0 +1,277 @@
+    			if (it.getAssemblyContext_AllocationContext().getEncapsulatedComponent__AssemblyContext() instanceof CompositeComponent) {
+    				resolvedAssignees.addAll(getComposite(assignments, context, (CompositeComponent) it.getAssemblyContext_AllocationContext().getEncapsulatedComponent__AssemblyContext()));
+    			}
+    		});
+    	return resolvedAssignees;
+	}
+	
+	/**
+	 * Gets the assembly assignees of a aSEFF node
+	 * @param assignments Resolved assignment container
+	 * @param assemblyContext Given assembly context
+	 * @return List of resolved assignees matching the node
+	 */
+	private List<AbstractAssignee> getAllocation(Assignments assignments, AssemblyContext assemblyContext) {
+		return assignments.getAssignee().stream()
+				.filter(AssemblyAssignee.class::isInstance)
+				.map(AssemblyAssignee.class::cast)
+				.filter(it -> it.getAssemblycontext().equals(assemblyContext))
+				.collect(Collectors.toList());
+	}
+	
+	/**
+	 * Gets the resource assignees of a SEFF node
+	 * @param assignments Resolved assignment container
+	 * @param resourceContainer Given resource container
+	 * @return List of resolved assignees matching the node
+	 */
+	private List<AbstractAssignee> getResource(Assignments assignments, ResourceContainer resourceContainer) {
+		return assignments.getAssignee().stream()
+				.filter(RessourceAssignee.class::isInstance)
+				.map(RessourceAssignee.class::cast)
+				.filter(it -> it.getResourcecontainer().equals(resourceContainer))
+				.collect(Collectors.toList());
+	}
+	
+	/**
+	 * Gets the contained assembly assignees of a SEFF node in a composite component
+	 * @param assignments Resolved assignment container
+	 * @param context Context of the node
+	 * @param compositeComponent Given composite component
+	 * @return
+	 */
+	private List<AbstractAssignee> getComposite(Assignments assignments, Deque<AssemblyContext> context, CompositeComponent compositeComponent) {
+		List<AbstractAssignee> evaluatedAssignees = new ArrayList<>();
+		List<AssemblyContext> assemblyContexts = compositeComponent.getAssemblyContexts__ComposedStructure();
+		for (AssemblyContext assemblyContext : assemblyContexts) {
+			if (context.contains(assemblyContext)) {
+				evaluatedAssignees.addAll(this.getAllocation(assignments, assemblyContext));
+			}
+		}
+		return evaluatedAssignees;
+	}
+
+	/**
+	 * Resolves the assignment container in the list of loaded resources
+	 * @return Assignment container of the model, or a dummy container, if no assignments exist
+	 */
+	private Assignments resolveAssignments() {
+		return this.resourceLoader.lookupElementOfType(NodeCharacteristicsPackage.eINSTANCE.getAssignments())
+	            .stream()
+	            .filter(Assignments.class::isInstance)
+	            .map(Assignments.class::cast)
+	            .findFirst().orElse(NodeCharacteristicsFactory.eINSTANCE.createAssignments());
+	}
+	
+	public void checkAssignments(Assignments assignments) {
```

Comment:
We should discuss the public API of this class. Its super class only required one method, however, several more are public. Are they required outside this class?